python/ray/serve/batching.py

```diff
@@ -250,15 +251,23 @@ async def _assign_func_results(
     async def _process_batches(self, func: Callable) -> None:
         """Loops infinitely and processes queued request batches."""
 
-        while True:
+        while not self._loop.is_closed():
             try:
                 self.curr_iteration_start_time = time.time()
                 await self._process_batch(func)
             except Exception:
                 logger.exception(
-                    "_process_batches asyncio task ran into an " "unexpected exception."
+                    "_process_batches asyncio task ran into an unexpected exception."
                 )
 
+        logger.warning(
```

Comment:
I don't think this is needed and might spam in peoples' logs during shutdown. at a minimum, make sure it's only logged to the file

Reply:
I removed the log. Initially, I though it would be useful because it'll only get logged once per replica when Serve is shutting down.

But I just realized that it'll also get logged whenever replicas are shutting down, which includes autoscaling.

Reply:
How spammy would it be? I think there's still value of knowing if we exited it prematurely

Reply:
We should not log it as a warning tho